beacon_node/network/src/network_beacon_processor/sync_methods.rs

```diff
@@ -37,11 +39,66 @@ pub enum ChainSegmentProcessId {
 }
 
 /// Returned when a chain segment import fails.
-struct ChainSegmentFailed {
+#[derive(Debug)]
+pub struct ChainSegmentFailed {
     /// To be displayed in logs.
-    message: String,
+    pub message: String,
     /// Used to penalize peers.
-    peer_action: Option<PeerAction>,
+    pub peer_action: Option<PeerGroupAction>,
+}
+
+/// Tracks which block(s) component caused the block to be invalid. Used to attribute fault in sync.
+#[derive(Debug)]
+pub struct PeerGroupAction {
+    pub block_peer: Option<PeerAction>,
+    pub column_peer: HashMap<ColumnIndex, PeerAction>,
+}
+
+impl PeerGroupAction {
+    fn block_peer(action: PeerAction) -> Self {
+        Self {
+            block_peer: Some(action),
+            column_peer: <_>::default(),
+        }
+    }
+
+    fn column_peers(columns: &[ColumnIndex], action: PeerAction) -> Self {
+        Self {
+            block_peer: None,
+            column_peer: HashMap::from_iter(columns.iter().map(|index| (*index, action))),
+        }
+    }
+
+    fn from_availability_check_error(e: &AvailabilityCheckError) -> Option<Self> {
+        match e {
+            AvailabilityCheckError::InvalidBlobs(_) => {
+                Some(PeerGroupAction::block_peer(PeerAction::LowToleranceError))
+            }
+            AvailabilityCheckError::InvalidColumn(errors) => Some(PeerGroupAction::column_peers(
+                &errors.iter().map(|(index, _)| *index).collect::<Vec<_>>(),
+                PeerAction::LowToleranceError,
+            )),
+            AvailabilityCheckError::ReconstructColumnsError(_) => None, // internal error
```

Comment:
We categorise this as `Malicious` in `AvailabilityCheckError::category`

Reply:
Wonder if we could reuse `AvailabilityCheckError::category` as we use that for block lookups and gossip message penalties.

Reply:
Yes we could, but we should move it outside of the beacon_chain crate as we don't have access to PeerAction there

Reply:
Removed `ReconstructColumnsError` for `Unexpected`

Reply:
`ReconstructColumnsError` may not be unexpected in the future if we start doing reconstruction on sync right?

Reply:
It will still be unexpected. If we verify the columns to be valid, and have >= 50% reconstruction should always succeed. A reconstruction error is something wrong with the underlying crypto lib AFAIK